app/src/main/java/com/android/bookswap/ui/chat/ChatScreen.kt

```diff
@@ -173,6 +175,56 @@ fun ChatScreen(
         }
       }
       delay(2000) // Delay for 2 seconds
+
+      // Check and add contacts for both users
+      if (messages.isNotEmpty()) {
+        // Add otherUser to currentUser's contacts
+        userSource.getUser(currentUser.userUUID) { result ->
+          if (result.isSuccess) {
+            val updatedUser = result.getOrThrow()
+            if (!updatedUser.contactList.contains(otherUser.userUUID.toString())) {
+              userSource.addContact(currentUser.userUUID, otherUser.userUUID.toString()) {
+                  contactResult ->
+                if (contactResult.isSuccess) {
+                  Log.d(
+                      "ChatScreen",
+                      "Added ${otherUser.userUUID} to ${currentUser.userUUID}'s contacts")
+                } else {
+                  Log.e(
+                      "ChatScreen",
+                      "Failed to add contact: ${contactResult.exceptionOrNull()?.message}")
+                }
+              }
+            }
+          } else {
+            Log.e(
+                "ChatScreen", "Failed to fetch current user: ${result.exceptionOrNull()?.message}")
+          }
+        }

```

Comment:
The user currently logged in the app is already stored in the UserViewModel so this call can most likely be simplified as `UserViewModel.getUser(force = true)`
On top of that, the current implementation doesn't comply with the MVVM template as this is calling data from the UI.

Reply:
Next week, when I fix the listener, I will change the whole OnLaunchEffect, I can do this then